draft-ietf-core-object-security.md

```diff
@@ -928,7 +925,7 @@ The operation of validating the Partial IV and updating the replay protection MU
 
 The following applies additionally when Observe is supported.
 
-The Notification Number is initialized to the Partial IV of the first successfully verified notification in response to the registration request. A client receiving a notification SHALL compare the Partial IV with the Notification Number associated to that Observe registration. The client MUST stop processing notifications with a Partial IV which has been previously received. Applications MAY decide that a client only processes notifications which have greater Partial IV than the Notification Number.
+The Notification Number is initialized to the Partial IV of the first successfully verified notification in response to the registration request, or to zero if the first successfully verified notification does not contain a Partial IV. A client MUST only accept at most one Observe notifications without Partial IV, and treat it as the oldest notification received. A client receiving a notification containing a Partial IV SHALL compare the Partial IV with the Notification Number associated to that Observe registration. The client MUST stop processing notifications with a Partial IV which has been previously received. Applications MAY decide that a client only processes notifications which have greater Partial IV than the Notification Number.
```

Comment:
Zero: should be "negative one". It'd only matter in the fringe case where the first request to a fresh server is an observation: With absent equals zero, the first actual notification (with PIV=0) would be disregarded by the client. (And PIVs can start at 0, cf. "except in the case of Partial IV of value 0 which is encoded to the byte string 0x00").

Reply:
That's correct, I missed that! Actually, the change we were discussing (and I thought I implemented -.-) was to remove the following: ", or to zero if the first successfully verified notification does not contain a Partial IV". I.e. the Notification Number would only be initialized after the first Partial IV is received (note: an implementation could set it to something like a negative integer, but that is implementation specific). In fact, Notification Number is only used after then, because the following sentence " A client MUST only accept at most one Observe notifications without Partial IV, and treat it as the oldest notification received" covers the replay and Observe processing of the one without Partial IV... would that be sufficient for you?